### PR TITLE
Show cli help as rendered markdown not as plain text

### PR DESCRIPTION
**Pull request by plaindocs**
I'm kinda excited about this one. :tada: 

Reviewers:

- [x] suggestions on better approaches for the ruby or script welcome

Still do to: 

- [ ] might need to tweak how we show cli options (what do we use elsewhere
- [x] double check all headings still work

# Before 
image

# After
image

### CODE REVIEW

**Comment by toolmantim** [comment]
I didn’t know how it would go rendering this to markdown, but I love it! 😍

**Review by toolmantim — APPROVED** [review]
Hey, if the Ruby works! 👍🏼 Might be worth renaming the files to `.md` from `.txt` now tho?

**Comment by plaindocs** [comment]
Yeah, we do need to promote all H2 to H3 in pages where we include multiple "man" pages

```
  "artifact download"
  "artifact shasum"
  "meta-data exists"
  "meta-data get"
  "meta-data keys"
  "meta-data set"
  "step get"
  "step update"
```

File: scripts/cli2md.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        puts "`#{line.strip}`"      
+    # Code sections
+    elsif /\s{3}\$/ =~ line
+        puts " #{line}"
+    # Lists of parameters
+    #    --config value                         Path to a configuration file [$BUILDKITE_AGENT_CONFIG]
+    elsif /\s{3}(-{2}[a-z0-9\- ]*)([A-Z].*)$/ =~ line
+        command = $1.rstrip
+        desc    = $2
+        desc.gsub!(/(\$BUILDKITE[A-Z0-1_]*)/,"`\\1`")
```
Comment (ticky) [inline scripts/cli2md.rb:16]: My gut says this might be a typo:

```suggestion
        desc.gsub!(/(\$BUILDKITE[A-Z0-9_]*)/,"`\\1`")
```

Comment (ticky) [reply scripts/cli2md.rb:16]: (the --no-http2 flag shows the issue with the current implementation)

File: scripts/cli2md.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+ARGF.each_with_index do |line, line_num|
+    # Headings 
+    if /^([A-z]*):/ =~ line
```
Comment (ticky) [inline scripts/cli2md.rb:3]: Might be better to just use the "word character" selector here; `A-z` includes a bunch of other characters which aren't alphabetical:

```suggestion
    if /^(\w*):/ =~ line
```

File: scripts/cli2md.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    elsif /\s{3}\$/ =~ line
+        puts " #{line}"
+    # Lists of parameters
+    #    --config value                         Path to a configuration file [$BUILDKITE_AGENT_CONFIG]
+    elsif /\s{3}(-{2}[a-z0-9\- ]*)([A-Z].*)$/ =~ line
+        command = $1.rstrip
+        desc    = $2
+        desc.gsub!(/(\$BUILDKITE[A-Z0-1_]*)/,"`\\1`")
+        puts "* `#{command}` - #{desc}"
+    elsif
```
Comment (ticky) [inline scripts/cli2md.rb:18]: Should this just be an else? I'm surprised permits an `elsif` without a condition!

```suggestion
    else
```

**Review by ticky — APPROVED** [review]
Looks great! A couple of little code issues but otherwise it's a great little upgrade :)

File: scripts/cli2md.rb (hunk 1 above)
Comment (plaindocs) [reply scripts/cli2md.rb:16]: Ah yes, I fixed that but didn't commit. Took me a while though. Good eyes.

**Comment by plaindocs** [comment]
Thanks @ticky, that is super helpful.

**Comment by plaindocs** [comment]
@ticky would you mind taking another look? If you can suggest a better way of demoting headings in a subset of the files that would be great. The "meta-data" page  for examples, includes 

```
../pages/agent/v3/cli_meta_data.md.erb
13:<%= render 'agent/v3/help/meta_data_set.md' %>
21:<%= render 'agent/v3/help/meta_data_get.md' %>
25:<%= render 'agent/v3/help/meta_data_exists.md' %>
29:<%= render 'agent/v3/help/meta_data_keys.md' %>
```

so the headings need to be H3 not H2 for the page structure to work as is.

**Comment by ticky** [comment]
Hmm, are those used in multiple places or just one?

**Comment by plaindocs** [comment]
@ticky each include is included once, but three of them are included in `/agent/v3/cli_meta_data`, basically, here is the meta-data page, you can set, get, check if it exists and look at keys....

**Comment by ticky** [comment]
Yeah, I think my other thought on how you could approach it is to make the CLI output to Markdown conversion happen on demand rather than manually, and leave the files as txt. Then you can add a parameter to the helper when called in the view to tell it how far to indent. It may be overly optimising though, I’m not sure it’s better than the sed solution if it works! 😂

**Comment by plaindocs** [comment]
Yeah, I tried that over in #975 but couldn't quite make it work. I think I'm just gonna merge it for now, as good enough. :+1:

**Pull request merged**